stdlib/public/core/EmptyCollection.swift

```diff
@@ -31,12 +31,16 @@ public struct EmptyIterator<Element> : IteratorProtocol, Sequence {
 }
 
 /// A collection whose element type is `Element` but that is always empty.
-public struct EmptyCollection<Element> : Collection {
+public struct EmptyCollection<Element> :
+  RandomAccessCollection
+  // FIXME: swift-3-indexing-model - conform to MutableCollection as well
+{
   /// A type that represents a valid position in the collection.
   ///
   /// Valid indices consist of the position of every element and a
   /// "past the end" position that's not valid for use as a subscript.
   public typealias Index = Int
+  public typealias IndexDistance = Int
```

Comment:
I see the use of type alias in EmptyCollection while other collections appear to often depend on type inference. What is the preferred thing to do? I kinda like the use of explicit type alias. It provides a central location to change your mind and it aligns with the associated type it overlays.